utils/s3/client.hh

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <seastar/core/file.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/core/shared_ptr.hh>
+#include <seastar/http/client.hh>
+
+using namespace seastar;
+
+namespace s3 {
+
+struct range {
+    uint64_t off;
+    size_t len;
+};
+
+class client : public enable_shared_from_this<client> {
+    class upload_sink;
+    class readable_file;
+    sstring _host;
+    http::experimental::client _http;
+
+    struct private_tag {};
+    future<std::vector<temporary_buffer<char>>> get_object(sstring object_name, std::optional<std::pair<uint64_t, size_t>> range);
+
+public:
+    explicit client(ipv4_addr addr, private_tag);
+    static shared_ptr<client> make(ipv4_addr addr);
+
+    future<uint64_t> get_object_size(sstring object_name);
+    future<temporary_buffer<char>> get_object_contiguous(sstring object_name, std::optional<range> range = {});
+    future<> put_object(sstring object_name, temporary_buffer<char> buf);
```

Comment:
I see put_object is used only in tests. We do not support writing sstables to s3 yet in this version?

Reply:
Writing sstables is implemented in the next patch that adds uploading sink. PUT-ing an object is no-go for sstables as it needs knowing exact sstable size in advance